src/eth/storage/permanent_storage.rs

```diff
@@ -0,0 +1,109 @@
+
+use super::EthStorageError;
+use crate::eth::primitives::Account;
+use crate::eth::primitives::Address;
+use crate::eth::primitives::Block;
+use crate::eth::primitives::BlockNumber;
+use crate::eth::primitives::BlockSelection;
+use crate::eth::primitives::Execution;
+use crate::eth::primitives::ExecutionConflicts;
+use crate::eth::primitives::Hash;
+use crate::eth::primitives::LogFilter;
+use crate::eth::primitives::LogMined;
+use crate::eth::primitives::Slot;
+use crate::eth::primitives::SlotIndex;
+use crate::eth::primitives::StoragePointInTime;
+use crate::eth::primitives::TransactionMined;
+
+/// Permanent (committed) storage operations
+// TODO: add Metrified method
+#[async_trait]
+pub trait PermanentStorage: Send + Sync {
+    // Retrieves the last mined block number.
+    async fn read_current_block_number(&self) -> anyhow::Result<BlockNumber>;
+
+    /// Atomically increments the block number, returning the new value.
+    async fn increment_block_number(&self) -> anyhow::Result<BlockNumber>;
+    /// Checks if the transaction execution conflicts with the current storage state.
+    async fn check_conflicts(&self, execution: &Execution) -> anyhow::Result<Option<ExecutionConflicts>>;
+
+    /// Retrieves an account from the storage. Returns Option when not found.
+    async fn maybe_read_account(&self, address: &Address, point_in_time: &StoragePointInTime) -> anyhow::Result<Option<Account>>;
+
+    /// Retrieves an slot from the storage. Returns Option when not found.
+    async fn maybe_read_slot(&self, address: &Address, slot_index: &SlotIndex, point_in_time: &StoragePointInTime) -> anyhow::Result<Option<Slot>>;
+
+    /// Retrieves a block from the storage.
+    async fn read_block(&self, block_selection: &BlockSelection) -> anyhow::Result<Option<Block>>;
+
+    /// Retrieves a transaction from the storage.
+    async fn read_mined_transaction(&self, hash: &Hash) -> anyhow::Result<Option<TransactionMined>>;
+
+    /// Retrieves logs from the storage.
+    async fn read_logs(&self, filter: &LogFilter) -> anyhow::Result<Vec<LogMined>>;
+
+    /// Persists atomically all changes from a block.
+    async fn save_block(&self, block: Block) -> anyhow::Result<(), EthStorageError>;
+
+    /// Persists initial accounts (test accounts or genesis accounts).
+    async fn save_initial_accounts(&self, accounts: Vec<Account>) -> anyhow::Result<()>;
+
+    /// Temporarily stores account changes during block production
+    async fn save_account_changes(&self, block_number: BlockNumber, execution: Execution) -> anyhow::Result<()>;
+
+    /// Resets all state to a specific block number.
+    async fn reset(&self, number: BlockNumber) -> anyhow::Result<()>;
+    /// Enables genesis block.
+    ///
+    /// TODO: maybe can use save_block from a default method.
+    async fn enable_genesis(&self, genesis: Block) -> anyhow::Result<()>;
+
+    // -------------------------------------------------------------------------
+    // Default operations
```

Comment:
All default operations can be moved to `StratusStorage` struct instead of being part of a trait.